ev3dev2/motor.py

```diff
@@ -67,105 +67,126 @@
     raise Exception("Unsupported platform '%s'" % platform)
 
 
-class SpeedInteger(int):
+class SpeedMeasure():
     """
     A base class for other unit types. Don't use this directly; instead, see
     :class:`SpeedPercent`, :class:`SpeedRPS`, :class:`SpeedRPM`,
     :class:`SpeedDPS`, and :class:`SpeedDPM`.
     """
     pass
 
-class SpeedPercent(SpeedInteger):
+class SpeedPercent(SpeedMeasure):
     """
     Speed as a percentage of the motor's maximum rated speed.
     """
 
+    def __init__(self, percent):
+        assert -100 <= percent <= 100,\
+            "{} is an invalid percentage, must be between -100 and 100 (inclusive)".format(percent)
+        
+        self.percent = percent
+
     def __str__(self):
-        return int.__str__(self) + "%"
+        return str(self.percent) + "%"
 
-    def get_speed_pct(self, motor):
+    def to_native_units(self, motor):
         """
-        Return the motor speed percentage represented by this SpeedPercent
+        Return this SpeedPercent in native motor units
         """
-        return self
+        return self.percent / 100 * motor.max_speed
 
 
-class SpeedNativeUnits(SpeedInteger):
+class SpeedNativeUnits(SpeedMeasure):
```

Comment:
Just a thought: there are `SpeedRPS`, `SpeedDPS`, etc. Why not call this `SpeedCPS` for the sake of uniformity? I understand that this class is special in the sense it *is* native, so I am not insisting.

Reply:
Google says "CPS" is cost-per-sale :). But I did need to look into comments to understand what DPS or DPM was anyway. What do think of `SpeedUnits` instead of `SpeedNativeUnits`?

Reply:
I feel like `SpeedUnits` would indicate that it's the base class without any particular units. Perhaps we could simplify this problem and call it `SpeedNative` -- I feel like that communicates that it's "whatever the motor measures in" while still being short.

Reply:
`MotorSpeedUnits`? Anyway, if people will mostly use other convenience classes and won't have to type the name of this class all the time, then the point is probably moot.